test/integration-new/ipamd/warm_target_test_PD_enabled.go

```diff
@@ -0,0 +1,353 @@
+				utils.AwsNodeName, utils.AwsNodeNamespace, utils.AwsNodeName,
+				map[string]string{
+					"WARM_IP_TARGET":    strconv.Itoa(warmIPTarget),
+					"MINIMUM_IP_TARGET": strconv.Itoa(minIPTarget),
+					"ENABLE_PREFIX_DELEGATION": "true",
+				})
+
+			// Allow for IPAMD to reconcile it's state
+			time.Sleep(utils.PollIntervalLong)
+
+			// Query the EC2 Instance to get the list of available Prefixes on the instance
+			primaryInstance, err = f.CloudServices.
+				EC2().DescribeInstance(*primaryInstance.InstanceId)
+			Expect(err).ToNot(HaveOccurred())
+
+			//Query for coredns pods
+	        podList, perr := f.K8sResourceManagers.PodManager().
+				GetPodsWithLabelSelector(podLabelKey, podLabelVal)
+			Expect(perr).ToNot(HaveOccurred())
+
+			assigned := 0
+			for _, pod := range podList.Items {
+				By(fmt.Sprintf("verifying in node %s but pod's IP %s address belongs to node name %s",
+					*primaryInstance.PrivateDnsName, pod.Status.PodIP, pod.Spec.NodeName))	
+				if pod.Spec.NodeName == *primaryInstance.PrivateDnsName {
+					assigned++
+					break
+				}
+			}
+
+			// Sum all the IPs on all network interfaces minus the primary IPv4 address per ENI
+			for _, networkInterface := range primaryInstance.NetworkInterfaces {
+				availPrefixes += len(networkInterface.Ipv4Prefixes)
+			}
+
+			// Validated avail IP equals the warm IP Size
+			prefixNeededForWarmIPTarget := ceil(assigned + warmIPTarget, 16)
+			prefixNeededForMinIPTarget := ceil(minIPTarget, 16)
+			Expect(availPrefixes).Should(Equal(Max(prefixNeededForWarmIPTarget, prefixNeededForMinIPTarget)))
+		})
+
+		JustAfterEach(func() {
+			k8sUtils.RemoveVarFromDaemonSetAndWaitTillUpdated(f,
+				utils.AwsNodeName, utils.AwsNodeNamespace, utils.AwsNodeName,
+				map[string]struct{}{
+					"WARM_IP_TARGET": {}, 
+					"MINIMUM_IP_TARGET": {},
+					"ENABLE_PREFIX_DELEGATION": {},
+				})
+		})
+
+		Context("when WARM_IP_TARGET = 2", func() {
+			BeforeEach(func() {
+				warmIPTarget = 2
+				minIPTarget = 0
+			})
+
+			It("should have 1 prefix", func() {})
+		})
+
+		Context("when WARM_IP_TARGET = 16", func() {
+			BeforeEach(func() {
+				warmIPTarget = 16
+				minIPTarget = 0
+			})
+
+			It("should have 1 prefix", func() {})
+		})
+
+		Context("when MINIMUM_IP_TARGET = 2", func() {
+			BeforeEach(func() {
+				warmIPTarget = 0
+				minIPTarget = 2
+			})
+
+			It("should have 1 prefix", func() {})
+		})
+
+		Context("when MINIMUM_IP_TARGET = 16", func() {
+			BeforeEach(func() {
+				warmIPTarget = 0
+				minIPTarget = 16
+			})
+
+			It("should have 1 prefixe", func() {})
+		})
+
+		Context("when MINIMUM_IP_TARGET = 6 and WARM_IP_TARGET = 10", func() {
+			BeforeEach(func() {
+				warmIPTarget = 6
+				minIPTarget = 10
+			})
+
+			It("should have 1 prefix", func() {})
+		})
+	})
+
+	Context("when warm prefix target is set with PD enabled", func() {
+		var warmPrefixTarget int
+
+		JustBeforeEach(func() {
+			var availPrefixes int
+			podLabelKey := "eks.amazonaws.com/component"
```

Comment:
Yeah I think it is better to make it more dynamic since mainly we need to check if these function - https://github.com/aws/amazon-vpc-cni-k8s/blob/master/pkg/ipamd/ipamd.go#L1273 and https://github.com/aws/amazon-vpc-cni-k8s/blob/62f1e7a84d1ef3e8d62d82b37f209cf57e0ccadb/pkg/ipamd/ipamd.go#L529-L539 provide expected o/p. I can discuss with you offline more.